python/lsst/pipe/base/mermaid_tools.py

```diff
@@ -0,0 +1,431 @@
+
+    Parameters
+    ----------
+    task_id : str
+        Unique Mermaid node identifier for this task.
+    taskDef : TaskDef
+        The pipeline task definition, which includes the task label, task name,
+        and connections.
+    universe : DimensionUniverse
+        Used to conform and sort the task's dimensions.
+    file : file-like
+        The output file-like object to write the Mermaid node definition.
+    show_dimensions : bool
+        If True, display the task's dimensions after conforming them.
+    """
+    # Basic info: bold label, then task name.
+    lines = [
+        f"<b>{html.escape(taskDef.label)}</b>",
+        html.escape(taskDef.taskName),
+    ]
+
+    # If requested, display the task's conformed dimensions.
+    if show_dimensions and taskDef.connections and taskDef.connections.dimensions:
+        task_dims = _expand_dimensions(taskDef.connections.dimensions, universe)
+        if task_dims:
+            dim_str = _format_dimensions(task_dims)
+            lines.append(f"<i>dimensions:</i>&nbsp;{html.escape(dim_str)}")
+
+    # Join with <br> for line breaks and define the node with the label.
+    label = "<br>".join(lines)
+    print(f'{task_id}["{label}"]', file=file)
+    print(f"class {task_id} task;", file=file)
+
+
+def _render_dataset_node(
+    ds_id: str,
+    ds_name: str,
+    connection: connectionTypes.BaseConnection,
+    universe: DimensionUniverse,
+    file: Any,
+    show_dimensions: bool,
+    show_storage: bool,
+) -> None:
+    """Render a dataset-type node in the Mermaid diagram.
+
+    Parameters
+    ----------
+    ds_id : str
+        Unique Mermaid node identifier for this dataset.
+    ds_name : str
+        The dataset type name.
+    connection : BaseConnection
+        The dataset connection object, potentially dimensioned and having a
+        storage class.
+    universe : DimensionUniverse
+        Used to conform and sort the dataset's dimensions if it is dimensioned.
+    file : file-like
+        The output file-like object to write the Mermaid node definition.
+    show_dimensions : bool
+        If True, display the dataset's conformed dimensions.
+    show_storage : bool
+        If True, display the dataset's storage class if available.
+    """
+    # Start with the dataset name in bold.
+    lines = [f"<b>{html.escape(ds_name)}</b>"]
+
+    # If dimensioned and requested, show conformed dimensions.
+    ds_dims = []
+    if show_dimensions and isinstance(connection, connectionTypes.DimensionedConnection):
+        ds_dims = _expand_dimensions(connection.dimensions, universe)
+
+    if ds_dims:
+        dim_str = _format_dimensions(ds_dims)
+        lines.append(f"<i>dimensions:</i>&nbsp;{html.escape(dim_str)}")
+
+    # If storage class is available and requested, display it.
+    if show_storage and getattr(connection, "storageClass", None) is not None:
+        lines.append(f"<i>storage class:</i>&nbsp;{html.escape(str(connection.storageClass))}")
```

Comment:
Adding a `&nbsp;` between `storage` and `class` would also help prevent line breaks here too.